### PR TITLE
feat: include API response headers in 'Last Response'

### PR DESCRIPTION
**Pull request by childish-sambino**
The default HTTP client stores 'Last Request' and 'Last Response' information for access to more data about the raw API request/response. The response object did not contain the headers until now.

### CODE REVIEW

File: src/main/java/com/twilio/http/ValidationClient.java [hunk 1]
```diff
@@ -84,7 +84,8 @@ public Response makeRequest(Request request) {
             HttpResponse response = client.execute(builder.build());
             return new Response(
                 response.getEntity() == null ? null : response.getEntity().getContent(),
-                response.getStatusLine().getStatusCode()
+                response.getStatusLine().getStatusCode(),
+                response.getAllHeaders()
```
Comment (eshanholtz) [inline src/main/java/com/twilio/http/ValidationClient.java:88]: Where's the definition for `getAllHeaders`? I only see `getHeaders`

Comment (childish-sambino) [reply src/main/java/com/twilio/http/ValidationClient.java:88]: https://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/HttpMessage.html#getAllHeaders()

Comment (eshanholtz) [reply src/main/java/com/twilio/http/ValidationClient.java:88]: Then what's the benefit of the `getHeaders` function?

Comment (childish-sambino) [reply src/main/java/com/twilio/http/ValidationClient.java:88]: `getHeaders` requires a string argument for getting specific headers.

Comment (eshanholtz) [reply src/main/java/com/twilio/http/ValidationClient.java:88]: [I don't see it taking in any arguments](https://github.com/twilio/twilio-java/pull/553/files/b43da0e4f047cf549bab5941d2387248012be505#diff-feb98a20878237b5c61d91e7f832f4f5R117)

Comment (childish-sambino) [reply src/main/java/com/twilio/http/ValidationClient.java:88]: `HttpResponse` != `Response`

**Pull request merged**